web/src/ApplicationListPage.js

```diff
@@ -123,7 +123,7 @@ class ApplicationListPage extends BaseListPage {
         render: (text, record, index) => {
           return (
             <Link to={`/applications/${record.organization}/${text}`}>
-              {text}
+              {record.isShared ? `${text}(shared)` : text}
```

Comment:
Define `Setting.getApplicationName(application)` to return your "XXX (Shared)" string.

"XXX (Shared)", not "XXX(shared)"